sei-db/ss/pebbledb/db.go

```diff
@@ -1085,3 +1204,130 @@ func retrieveLastRangeHashed(db *pebble.DB) (int64, error) {
 	}
 	return int64(ubz), nil
 }
+
+// collectMetricsInBackground periodically collects PebbleDB internal metrics
+func (db *Database) collectMetricsInBackground() {
+	ticker := time.NewTicker(10 * time.Second) // Collect metrics every 10 seconds
+	defer ticker.Stop()
+
+	for db.metricsCtx.Err() == nil {
+		select {
+		case <-db.metricsCtx.Done():
+			return
+		case <-ticker.C:
+			db.collectAndRecordMetrics(db.metricsCtx)
+		}
+	}
+}
+
+// collectAndRecordMetrics collects PebbleDB internal metrics and records them
+func (db *Database) collectAndRecordMetrics(ctx context.Context) {
+	if db.storage == nil {
+		return
+	}
+
+	m := db.storage.Metrics()
+
+	db.metricsMu.Lock()
+	lastSnapshot := db.lastMetricsValues
+	db.metricsMu.Unlock()
+
+	// Compaction metrics - PebbleDB tracks compactions in the Compact field
+	compactionCount := m.Compact.Count
+
+	// Calculate bytes read/written from level metrics
+	var compactionBytesRead int64
+	var compactionBytesWritten int64
+	for level := 0; level < len(m.Levels); level++ {
+		levelMetrics := m.Levels[level]
+		compactionBytesRead += int64(levelMetrics.BytesIn)
+		compactionBytesWritten += int64(levelMetrics.BytesCompacted)
+	}
+
+	// Calculate deltas for counters
+	compactionCountDelta := compactionCount - lastSnapshot.compactionCount
```

Comment:
Why `Delta`? we can do all sorts of fancy aggregation in prometheus/grafana. if we simply return the count then we: 1) retain the ability to do higher quality aggregation in the metric reporting system outside of the application, and 2) simplify the implementation here by rendering `lastMetricsValues` unnecessary.

Ditto for the remaining such calculations.